tests/tests/t65-opdracht-delen.spec.ts

```diff
@@ -0,0 +1,211 @@
+import { expect, type Page, test } from '@playwright/test';
+
+import { cleanupTestdata } from '../helpers/cleanup';
+import { goToPageAndAcceptCookies } from '../helpers/go-to-page-and-accept-cookies';
+import { loginOnderwijsAvo } from '../helpers/login-onderwijs-avo';
+import { logoutOnderwijsAvo } from '../helpers/logout-onderwijs-avo';
+
+/**
+ * This test requires `PROXY_E2E=true`
+ * https://github.com/viaacode/avo2-proxy/commit/ae4e79f3b1c4826f5c81ba570b15d904990d43b7
+ * https://github.com/viaacode/avo2-proxy/commit/9b9b740fea4b66478d850da4de8a1d74da8d845e
+ */
+
+test.afterEach(async ({ page }, testInfo) => {
+	if (testInfo.status !== testInfo.expectedStatus) {
+		console.log(`Did not run as expected`);
+		await cleanupTestdata(page);
+	}
+});
+
+test('T65: opdracht delen van lesgever lager naar lesgever secundair', async ({ page }) => {
+	await shareAssignmentAndAccept(
+		page,
+		{
+			user: process.env.TEST_LESGEVER_LAGER_GEBRUIKER_USER as string,
+			pass: process.env.TEST_LESGEVER_LAGER_GEBRUIKER_PASS as string,
+		},
+		{
+			user: process.env.TEST_LESGEVER_SECUNDAIR_GEBRUIKER_USER as string,
+			pass: process.env.TEST_LESGEVER_SECUNDAIR_GEBRUIKER_PASS as string,
+		},
+		false
+	);
+});
+
+test('T65: opdracht delen van lesgever secundair naar lesgever lager', async ({ page }) => {
+	await shareAssignmentAndAccept(
+		page,
+		{
+			user: process.env.TEST_LESGEVER_SECUNDAIR_GEBRUIKER_USER as string,
+			pass: process.env.TEST_LESGEVER_SECUNDAIR_GEBRUIKER_PASS as string,
+		},
+		{
+			user: process.env.TEST_LESGEVER_LAGER_GEBRUIKER_USER as string,
+			pass: process.env.TEST_LESGEVER_LAGER_GEBRUIKER_PASS as string,
+		},
+		false
+	);
+});
+
+test('T65: opdracht delen van lesgever lager naar lesgever beide', async ({ page }) => {
+	await shareAssignmentAndAccept(
+		page,
+		{
+			user: process.env.TEST_LESGEVER_LAGER_GEBRUIKER_USER as string,
+			pass: process.env.TEST_LESGEVER_LAGER_GEBRUIKER_PASS as string,
+		},
+		{
+			user: process.env.TEST_LESGEVER_BEIDE_GEBRUIKER_USER as string,
+			pass: process.env.TEST_LESGEVER_BEIDE_GEBRUIKER_PASS as string,
+		},
+		true
+	);
+});
+
+test('T65: opdracht delen van lesgever secundair naar lesgever beide', async ({ page }) => {
+	await shareAssignmentAndAccept(
+		page,
+		{
+			user: process.env.TEST_LESGEVER_SECUNDAIR_GEBRUIKER_USER as string,
+			pass: process.env.TEST_LESGEVER_SECUNDAIR_GEBRUIKER_PASS as string,
+		},
+		{
+			user: process.env.TEST_LESGEVER_BEIDE_GEBRUIKER_USER as string,
+			pass: process.env.TEST_LESGEVER_BEIDE_GEBRUIKER_PASS as string,
+		},
+		true
+	);
+});
+
+async function createAssignmentWithTitle(page: Page) {
```

Comment:
shouldn't these helpers be moved to the helpers folder
or will they only be used inside this test?

Reply:
Lots of tests use slightly similar code. I split it into declarative functions in this file for my own sanity but I don't think we can easily implement these as helpers in other files.